app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,39 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+    @current_person = @user
+  end
```

Comment:
- [ ] Let's ensure that we solve the `n + 1 query` problem by including the comments when fetching a user's posts, this way, we only get to query the database once when fetching posts because we included the comments on that query. But when you do not include the comments, the database is queried based on the number of comments the user has which results in a `n + 1 query` problem. So let's solve it by replacing line `4` with `@posts = @user.posts.includes(:comments)`. You can check this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh#:~:text=What%20is%20the%20N%2B1,the%20result%20of%20that%20data.) for more explanation on `n + 1 query` problem in rails and how to solve it.